gson/src/test/java/com/google/gson/functional/SerializedNameTest.java

```diff
@@ -51,4 +53,26 @@ public void testMultipleNamesInTheSameString() {
       this.b = b;
     }
   }
+
+  public void testPatternForEnum() {
+    Gson gson = new Gson();
+    assertTrue(Pattern.compile("\\S*").matcher("REDIRECT").matches()); // verify that the pattern is correct 
+    Response target = gson.fromJson("{'error1':'AUTH_FAILURE','error2':'REDIRECT','error3':'SERVER_ERROR'}", Response.class);
+    assertEquals(ErrorCode.AUTH_FAILURE, target.error1);
+    assertEquals(ErrorCode.UNKNOWN, target.error2);
+    assertEquals(ErrorCode.SERVER_ERROR, target.error3);
+  }
+
+  private enum ErrorCode {
+    AUTH_FAILURE,
+    SERVER_ERROR,
+    /** match all non-whitespace characters. Note that pattern is applied last. */
+    @SerializedName(value="UNKNOWN", pattern="\\S*") UNKNOWN
```

Comment:
We know the target enum ’cause you can call `e.getDeclaringClass()`. To implement this, we’d just install a type adapter like `EnumWithFallbackAdapter`. Ie. this would be sugar for:

```
Gson gson = new GsonBuilder()
    .registerTypeAdapter(ErrorCode.class, new EnumFallbackAdapter(ErrorCode.UNKNOWN))
    .create();

```